### ISSUE TITLE
[RFE] Support to optionally add watches during the testing

### ISSUE THREAD

**Issue by josecastillolema** [kube-burner/kube-burner#647]
**Is your feature request related to a problem? Please describe.**
Kubernetes supports efficient change notifications on resources via watches. Several k8s applications leverage them, i.e.: ArgoCD, Knative, etc. Watches are known to add load to KAPI in some scenarios.

**Describe the solution you'd like**
While kube-burner creates a watch for all the pods that are created during the testing, it would be nice to be able to optionally create additional watches for a number of resources (i.e.: namespaces with certain labels).

cc @williamcaban @smalleni @Noreen21

**Comment by rsevilla87** [comment]
Hey @josecastillolema , can you provide an example of what exactly are you going to get with this feature? is this related to create load in the API by adding watchers to some subset of objects?

**Comment by josecastillolema** [comment]
Exactly, @williamcaban was mentioning that some watchers are known to cause a severe overhead on KAPI resource consumption.

**Comment by williamcaban** [comment]
For more context, two actions are known to stress the K8s API: WATCH and LIST. In both cases, any K8s with a version < 1.30 (where we should see some fixes with API streaming). In the case of many LIST requests (which is what an application like Argo, or an audit tool does), it forces the API server to allocate enough memory to build the structure in memory to represent all the objects requested by the LIST action, which can be equivalent to all objects in etcd when is a LIST of all the objects in the cluster, to generate the answer to that request. Because of this, when a cluster has multiple applications or agents with regular large LIST requests, it tries to allocate all the available memory to the point where the control plane node resources are exhausted. There are features like APF that can indirectly help. The intent is to be able to replicate these unfavorable conditions to determine scalability, recoverability, and stability, which we can maintain in the control plane under stress.

In the case of WATCH, it directly impacts the scalability of how many Operators or custom controllers the system can handle. The intent of generating a WATCH load condition is to measure operator scalability, and stability under stressed conditions, etc. 

At the end, having both of these, WATCH and LIST, would also allow us to replicate conditions to the control plane that otherwise will require instantiating large number of workloads requiring large number of worker nodes (e.g., 8k Tekton pipelines, or 100 instances of Argo) when the fact is we need to generate the WATCH and LIST load, not to install the applications and create applications trying to achieve the same goal.
